File: src/test/samples/validSamples/validSample.ex
#region FirstRegion
x = 42
#endregion

# region Second Region
defmodule MyClass do
  #  region   InnerRegion
  def my_method do
  end
  #   endregion   ends InnerRegion

  #region
  def my_method2 do
  end
    #     endregion
end
#  endregion

File: src/test/samples/validSamples/validSample.cpp
#include <iostream>

//#region FirstRegion
int x = 42;
//#endregion

// #region Second Region
class MyClass {
    //  #region    InnerRegion    
    void myMethod() {}
    //   #endregion    ends InnerRegion  

    // #region
    void myMethod2() {}
    //#endregion
};
//  #endregion
